test/reader-test.js

```diff
@@ -127,6 +130,52 @@ function checkIPv6(reader){
   });
 });
 
+describe('Without IPv4 Search Tree', function(){
+  var reader = new Reader('test/data/MaxMind-DB-no-ipv4-search-tree.mmdb');
+
+  it('finds expected record for 1.1.1.1', function(){
+    assert.strictEqual(reader.lookup('1.1.1.1'), '::0/64');
+  });
+
+  it('finds expected record for 192.1.1.1', function(){
+    assert.strictEqual(reader.lookup('192.1.1.1'), '::0/64');
+  });
+
+});
+
+describe('Bad data', function(){
+
+  describe('Bad pointers', function(){
+    var reader = new Reader('test/data/MaxMind-DB-test-broken-pointers-24.mmdb');
+
+    it('throws with bad search tree', function(){
+      assert.throws(function(){
+        reader.lookup('1.1.1.32');
+      });
+    });
+
+    // I can't figure out why this doesn't throw. I _think_
+    // it's supposed to test whether a pointer can point to
+    // another pointer. Problem is, we don't throw on that.
+    // it('throws with bad data section', function(){
```

Comment:
I am not sure if it is related, but this test database was broken upstream for a while. It has since [been fixed](https://github.com/maxmind/MaxMind-DB/pull/15).